c10/cuda/CUDACachingAllocator.h

```diff
@@ -125,9 +128,39 @@ struct SegmentInfo {
   int64_t active_size = 0;
   cudaStream_t stream = 0;
   bool is_large = false;
+  MempoolId_t owner_private_pool_id = {0, 0};
```

Comment:
Tell me more? Why do segments need owner ids now?

Reply:
These are all debug properties. I'm using the private pool id in debugging (testing) so extending them in the structure